src/ruleset.rs

```diff
@@ -653,6 +653,36 @@ impl RulesetCreated {
         };
         Ok(body()?)
     }
+
+    /// Creates a copy of this object and duplicates the file descriptor with
+    /// CLOEXEC set. The `RulesetCreated` object that is produced is intended
+    /// to be a readonly copy of the original object, however this
+    /// is not currently supported by the landlock kernel module. The intention
+    /// behind this method is to provide a way to keep a parent ruleset intact so
+    /// it can be applied to any number of new processes or threads without
+    /// creating the object over and over.
+    ///
+    /// Any modification of the underlying ruleset changes will alter both
+    /// the new RulesetCreated and the original.
+    ///
+    /// On error, returns [`std::io::Error`].
+    pub fn try_clone(&self) -> Result<Self, std::io::Error> {
```

Comment:
Can you please return `Result<std::io::Error>` instead and update the test accordingly?

Reply:
Do you mean return a `std::io::Result<Self>`?